ports/stm32/boards/make-pins.py

```diff
@@ -355,7 +355,7 @@ def print(self):
     def print_adc(self, adc_num):
         print("")
         print("const pin_obj_t * const pin_adc{:d}[] = {{".format(adc_num))
-        for channel in range(17):
+        for channel in range(20):
```

Comment:
I'm not sure this is correct.  First of all it adds 3 entries for all MCUs, and most won't use these.  Second I don't think it works for H7 because it gives (on NUCLEO_H743ZI):
```c
const pin_obj_t * const pin_adc1[] = {
  &pin_A0_obj, // 0
  &pin_A1_obj, // 1
  NULL,    // 2
  &pin_A6_obj, // 3
  &pin_C4_obj, // 4
  &pin_B1_obj, // 5
  NULL,    // 6
  &pin_A7_obj, // 7
  &pin_C5_obj, // 8
  &pin_B0_obj, // 9
  &pin_C0_obj, // 10
  &pin_C1_obj, // 11
  &pin_C2_obj, // 12
  &pin_C3_obj, // 13
  &pin_A2_obj, // 14
  &pin_A3_obj, // 15
#if defined(STM32L4)
  NULL,    // 16
#endif
  NULL,    // 17
  &pin_A4_obj, // 18
  &pin_A5_obj, // 19
};
```
Simplest solution would be to just remove the `#if defined(STM32L4)` altogether, and use 20 entries for all MCUs (which wastes up to 3*4*4=48 bytes flash).  Complex solution would add more `#if`'s for H7 (and WB needs the 16th entry as well).

Reply:
Fixed properly.

Reply:
That said, I don't know I think maybe it's better to have 20 pins for all MCUs as future-proofing for new MCUs ? It's very easy to miss this when adding new MCUs. Or if there's a way to pass the max number of channels to `make-pins.py` that would be better, with a default in `mpconfigboard_common.h` like `MICROPY_HW_MAX_I2C` for example.

Reply:
I tried to fix this properly in #7477